src/api/layers/AbstractLayer.class.js

```diff
@@ -23,9 +23,11 @@ export default class AbstractLayer {
 
     /**
      * @abstract
+     * @param {Number} projection The EPSG number of the projection system to use (for instance,
+     *   EPSG:2056 will require an input of 2056)
      * @returns {String} The URL to use to request tile/image/data for this layer
      */
-    getURL() {
+    getURL(projection = 3857) {
```

Comment:
I would define all EPSG numbers as constants and use the constant name instead of those magic numbers, this will make the code much clearer. Always avoid magic numbers.
```javascript
EPSG_3857 = 3857
// or even better
EPSG_MERCATOR = 3857
WEB_MERCATOR = 3857
LV95 = 2056
...
```
And use the constant everywhere in the code instead of the hardcoded number

Reply:
So I've re-used what was declared in `coordinateUtils.js`, there was already an enum that was only missing the EPSG:3857 in it.
I think I've replaced all direct declaration of number or string regarding EPSG in the code with this enum